File: benches/lang_compare/fibonacci.cpp
// Fibonacci iterative - C++

#include <iostream>

long long fibonacci(int n) {
    if (n <= 1) return n;
    long long a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        long long temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    long long result = 0;
    for (int i = 0; i < 10000; i++) {
        result = fibonacci(1000);
    }
    return 0;
}

File: benches/lang_compare/fibonacci.rs
//! Fibonacci iterative - Rust

fn fibonacci(n: u64) -> u64 {
    if n <= 1 {
        return n;
    }
    let mut a = 0;
    let mut b = 1;
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    let mut result = 0;
    for _ in 0..10000 {
        result = fibonacci(1000);
    }
}
